sources/packages/api/src/services/application/application.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, Inject } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { Connection } from "typeorm";
+import { UserInfo } from "../../types";
+import { LoggerService } from "../../logger/logger.service";
+import { InjectLogger } from "../../common";
+import { Application } from "../../database/entities/application.model";
+import { CreateApplicationDto } from "../../route-controllers/application/models/application.model";
+import { StudentService } from "..";
+
+@Injectable()
+export class ApplicationService extends RecordDataModelService<Application> {
+  @InjectLogger()
+  logger: LoggerService;
+
+  constructor(
+    @Inject("Connection") connection: Connection,
+    private readonly studentService: StudentService,
+  ) {
+    super(connection.getRepository(Application));
+    this.logger.log("[Created]");
+  }
+
+  async createApplication(
+    userInfo: UserInfo,
+    applicationDto: CreateApplicationDto,
+  ): Promise<Application> {
+    // Find the student realted to the logged user.
+    const student = await this.studentService.getStudentByUserName(
+      userInfo.userName,
+    );
+
```

Comment:
If no student is found?

Reply:
At this point, the user is already authenticated and the user must exist in the DB, but I agree that it would be good to have a check here and maybe throw an exception with more context.

Reply:
Added a check and a better exception in the case something bad happens and the user is not present.